### PR TITLE
[3/N] Quantization Refactor: ModelSlim MoE schemes

### PR DESCRIPTION
**Pull request by TamirBaydasov**
## Motivation

Add MoE schemes to modelslim instead of storing all classes in a single file. Follow up to https://github.com/sgl-project/sglang/pull/17503

Images and motivation for this PR can be viewed in our roadmap: #15194

## Modifications
Moved all classes from `modelslim_moe.py` to new schemes in quantization/modelslim/schemes/
Removed `modelslim_moe.py` file
Added `get_moe_scheme` function to `modelslim.py` file

#### NPU tests:
Qwen3-30B-W8A8 (attn - static / mlp - dynamic)
image

## Benchmarking and Profiling
Not affected

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 1]
```diff
@@ -2,27 +2,36 @@
...
     ModelSlimScheme,
     ModelSlimW4A4Int4,
+    ModelSlimW4A8Int8MoE,
     ModelSlimW8A8Int8,
+    ModelSlimW8A8Int8MoE,
 )
 from sglang.srt.layers.quantization.unquant import UnquantizedLinearMethod
 from sglang.srt.utils import apply_module_patch
 
+if TYPE_CHECKING:
```
Comment (ping1jing2) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py:26]: please move `ModelSlimScheme` into TYPE_CHECKING block and check other import statements

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py [hunk 2]
```diff
@@ -274,3 +315,74 @@ def apply(
...
+        layer: torch.nn.Module,
+        num_experts: int,
+        hidden_size: int,
+        intermediate_size_per_partition: int,
+        params_dtype: torch.dtype,
+        **extra_weight_attrs,
+    ):
+        """
+        Use the ModelSlimScheme associated with each layer to create
+        the necessary parameters for the layer. See LinearMethodBase for param
```
Comment (ping1jing2) [inline python/sglang/srt/layers/quantization/modelslim/modelslim.py]: please double check if we should change `LinearMethodBase` into `FusedMoEMethodBase`

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py (hunk 1 above)
Comment (TamirBaydasov) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py:26]: Done

File: python/sglang/srt/layers/quantization/modelslim/modelslim.py (hunk 2 above)
Comment (TamirBaydasov) [reply python/sglang/srt/layers/quantization/modelslim/modelslim.py]: Done

**Review by ping1jing2 — APPROVED** [review]

**Review by AniZpZ — APPROVED** [review]
